hack/release/pkg/builder/builder.go

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builder
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/sirupsen/logrus"
+)
+
+// Global configuration for releases.
+var (
+	// Registries to which all release images are pushed.
+	registries = []string{
```

Comment:
This change requires https://github.com/projectcalico/calico/pull/5360 in order to work.

Currently we build images like `docker.io/calico/calico/kube-controllers` instead of `docker.io/calico/kube-controllers`